cetmix_tower_server/models/cx_tower_key_mixin.py

```diff
@@ -0,0 +1,45 @@
+from odoo import api, models
+
+
+class CxTowerKeyMixin(models.AbstractModel):
+    _name = "cx.tower.key.mixin"
+    _description = "Mixin for managing secrets"
+
+    @api.depends("code")
+    def _compute_secret_ids(self):
+        """
+        Compute the secret IDs based on the references found in the code field.
+
+        This method updates the secret_ids Many2many field by extracting secret
+        references from the code field. If no code is present, the field is cleared.
+        """
+        for record in self:
+            # Clear secrets by default
+            record.secret_ids = [(5, 0, 0)]
```

Comment:
This will always trigger an update operation, even if it's not needed.
You should check the difference between current secrets and new ones. Add modify this field only in case this difference exists.

Reply:
Thanks, understood